docs/spec.md

```diff
@@ -199,6 +199,19 @@ quantization parameters. Quantized tensor types have the following constraints:
   * (C12) `quantization_dimension < size(shape)`.
   * (C13) `size(scales) = shape[quantization_dimension]`.
 
+For simplicity, we only define individual operation semantics for per-tensor
+quantized type. Semantics of per-axis type, if supported by the op, can be
+derived as follows:
+
+Given an operation specification with per-tensor quantized type, let the
+semantics of the operation includes some application of the quantization
+parameters `zero_point` and `scale` to a tensor element`e` at index = `[i0, ...,
+id, ..., iR-1]`. The specification for the semantics of the same operation on
+tensor element `e`, with per-axis quantized type with quantization dimension `d`
+, would exactly be the same as the one with per-tensor type except that the
+`zero_point` and `scale` will be replaced with their per-axis counterparts,
+`zero_points[id]` and `scales[id]` resp.
```

Comment:
As discussed offline, I don't think this definition works well beyond elementwise ops. 

To be more specific: we already have `ReshapeOp`, `SliceOp` and `TransposeOp` which support per-axis quantization at HEAD, and this PR proposes to add support for per-axis quantization to `PadOp` and `ReverseOp`. 

For all of these ops, this definition is insufficient. For example, what if a reshape affects the quantization dimension? Or what if pad pads the quantization dimension? Similarly for other ops called out above. It doesn't appear that we can have a uniform way of deriving per-axis semantics from per-tensor semantics.

Let's delete these paragraphs from this PR and sort this out separately - in #1574.

Reply:
+1. Per-axis quantization (and mixed precision) is important for precision demanding tasks. Having a clear spec is important

Reply:
noted.